hw-model/src/model_emulated.rs

```diff
@@ -55,6 +55,21 @@ pub struct ModelEmulated {
     ready_for_fw: Rc<Cell<bool>>,
     cpu_enabled: Rc<Cell<bool>>,
     trace_path: Option<PathBuf>,
+
+    image_tag: u64,
+}
+impl Drop for ModelEmulated {
+    fn drop(&mut self) {
+        let cov_path =
+            std::env::var(caliptra_coverage::CPTRA_COVERAGE_PATH).unwrap_or_else(|_| "".into());
+        if cov_path.is_empty() {
+            return;
+        }
+
+        let bitmap = self.code_coverage_bitmap();
+        let _ =
```

Comment:
nitpick: Should probably print a message when this happens.

Reply:
follow-up pr.